src/register.cpp

```diff
@@ -52,25 +52,25 @@ int Register::registerNewType(
     throw std::invalid_argument("PyType_Ready fails to initialize input type.");
   }
 
-  PyArray_Descr* descr_ptr =
-      new PyArray_Descr(*call_PyArray_DescrFromType(NPY_OBJECT));
-  PyArray_Descr& descr = *descr_ptr;
-  descr.typeobj = py_type_ptr;
-  descr.kind = 'V';
-  descr.byteorder = '=';
-  descr.type = 'r';
-  descr.elsize = type_size;
-  descr.flags =
+  PyArray_DescrProto* proto_ptr = new PyArray_DescrProto();
+  PyArray_DescrProto& proto = *proto_ptr;
+  Py_SET_TYPE(&proto, &PyArrayDescr_Type);
+  proto.typeobj = py_type_ptr;
+  proto.kind = 'V';
+  proto.byteorder = '=';
+  proto.type = 'r';
+  proto.elsize = type_size;
+  proto.flags =
       NPY_NEEDS_PYAPI | NPY_USE_GETITEM | NPY_USE_SETITEM | NPY_NEEDS_INIT;
-  descr.type_num = 0;
-  descr.names = 0;
-  descr.fields = 0;
-  descr.alignment =
-      alignement;  // call_PyArray_DescrFromType(NPY_OBJECT)->alignment;
+  proto.type_num = 0;
+  proto.names = 0;
+  proto.fields = 0;
+  proto.alignment =
```

Comment:
Changing to `proto` does not improve the readability.

Reply:
Please revert it.

Reply:
ok. I was following numpy naming convention for proto-description of arrays.